sdk/core/azure-core/inc/azure/core/long_running_operation_state.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+#include <utility> // for std::move
+
+#include "azure/core/strings.hpp"
+
+/*
+Need two things:
+  Polling loop that does the wait for the customer
+  Enabling the scenario where the user can just wait
+*/
+
+namespace Azure { namespace Core {
+
+  class LongRunningOperationState {
+    std::string m_value;
+
+  public:
+    // Provide `explicit` conversion from string or types convertible to string:
+    explicit LongRunningOperationState(const std::string& value) : m_value(value) {}
```

Comment:
I have asked @antkmsft  what's the difference between 
`explicit LongRunningOperationState(const std::string& value) : m_value(value)`
vs
`explicit LongRunningOperationState(std::string value) : m_value(std::move(value))`

For what I understand, the second way provides a better and clearer intention about the method will make a copy of the string that we are passing.
While the first one, seems like the method is only reading the value (and we make a copy of it behind the scenes)

So, the preferred way should be the second approach

Reply:
Keep in mind, these are public APIs so we should try to be idiomatic and avoid unnecessary string copies.